### PR TITLE
Fix for #7147: Problems with use of big timeout (or no timeout at all) in trace service

### PR DESCRIPTION
**Pull request by AlexPeshkoff**

### CODE REVIEW

**Review by hvlad — COMMENTED** [review]
Does you consider to use already existing PortsCleanup ?

File: src/remote/client/interface.cpp [hunk 1]
```diff
@@ -995,6 +998,18 @@ class RProvider : public StdPlugin<IProviderImpl<RProvider, CheckStatusWrapper>
 void RProvider::shutdown(CheckStatusWrapper* status, unsigned int /*timeout*/, const int /*reason*/)
 {
 	status->init();
+
+	// make OS abort all outgoing connections
+	MutexLockGuard g(outPortsMutex, FB_FUNCTION);
+	for (auto p = outPorts->begin(); p != outPorts->end(); ++p)
```
Comment (asfernandes) [inline src/remote/client/interface.cpp:1004]: Why not `for (auto port : outPorts)`?

Comment (AlexPeshkoff) [reply src/remote/client/interface.cpp:1004]: It does not work with GlobalPtr<> variable.

**Comment by hvlad** [comment]
@AlexPeshkoff : could you explain what exact bug you going to fix and how ? I'm not sure I got it reading #7147 description.

**Comment by AlexPeshkoff** [comment]
The problem may happen with any service, querying server w/o timeout, as long as server itself never finishes connection with client. Most of our services do not behave this way - as soon as requested job is complete they return no data in Service::query() and finish this API call. But trace service behaves in another manner - when there is no data to return to the client it waits for timeout expiration or indefinitely when no timeout. I.e. in order to break connection with server client must be able to at least disconnect from it gracefully. The problem is that remote provider does nothing on an attempt to fb_shutdown() it. Therefore application which started Service::query() w/o timeout can't close connection calling fb_shutdown() and after using it waits for completion of Service::query() call. As long as server has enough data to return buffer full of data.

When working with trace service we were almost always using fbtracemgr & fbsvcmgr. Both tools use 1 second timeout when getting data from service. That was hiding mention effects for a long time. Looks like users were also always using timeout - may be looking at our code in fb*mgrs...

**Comment by AlexPeshkoff** [comment]
@hvlad Cancel happened to be really simple task - most of time took API modification. What do you think - do we want to support it remotely? Currently there is a stub in interface.cpp.

**Comment by hvlad** [comment]
> @hvlad Cancel happened to be really simple task - most of time took API modification. What do you think - do we want to support it remotely? Currently there is a stub in interface.cpp.

It is good to have, but not required for this ticket, imho.
If it could be implemented - why not ?

**Pull request merged**